lib/Doctrine/DBAL/Connections/MasterSlaveConnection.php

```diff
@@ -213,7 +213,24 @@ protected function connectTo($connectionName)
         $user = isset($connectionParams['user']) ? $connectionParams['user'] : null;
         $password = isset($connectionParams['password']) ? $connectionParams['password'] : null;
 
-        return $this->_driver->connect($connectionParams, $user, $password, $driverOptions);
+        if ($connectionName == 'slave') {
+            while(true) {
```

Comment:
Ah, I missed the `array_pop()`. Should be `do { } while ($connectionParams = array_pop($params['slaves']))` then